packages/main/src/DynamicDateRange.ts

```diff
@@ -0,0 +1,250 @@
+import "@ui5/webcomponents-icons/dist/appointment-2.js";
+import DynamicDateRangeTemplate from "./DynamicDateRangeTemplate.js";
+import IconMode from "./types/IconMode.js";
+import type Input from "./Input.js";
+import type { IDynamicDateRangeOption } from "./DynamicDateOption.js";
+import {
+	DYNAMIC_DATE_RANGE_SELECTED_TEXT,
+	DYNAMIC_DATE_RANGE_EMPTY_SELECTED_TEXT,
+} from "./generated/i18n/i18n-defaults.js";
+
+// default calendar for bundling
+import "@ui5/webcomponents-localization/dist/features/calendar/Gregorian.js";
+
+// Styles
+import dynamicDateRangeCss from "./generated/themes/DynamicDateRange.css.js";
+import dynamicDateRangePopoverCss from "./generated/themes/DynamicDateRangePopover.css.js";
+import ResponsivePopoverCommonCss from "./generated/themes/ResponsivePopoverCommon.css.js";
+
+type DynamicDateRangeValue = {
+	/**
+	 * The key of the option.
+	 * @default ""
+	 * @public
+	 */
+	operator: string;
+
+    /**
+     * Values of the dynamic date range.
+     * @default []
+     * @public
+     */
+    values?: Date[] | number[];
+}
+
+/**
+ * @class
+ * ### ES6 Module Import
+ *
+ * `import "@ui5/webcomponents/dist/DynamicDateRange.js";`
+ * @constructor
+ * @extends UI5Element
+ * @public
+ */
+
+@customElement({
+	tag: "ui5-dynamic-date-range",
+	languageAware: true,
+	template: DynamicDateRangeTemplate,
+	renderer: jsxRenderer,
+	styles: [
+		dynamicDateRangeCss,
+		ResponsivePopoverCommonCss,
+		dynamicDateRangePopoverCss,
+	],
+})
+
+class DynamicDateRange extends UI5Element {
+    @i18n("@ui5/webcomponents")
+	static i18nBundle: I18nBundle;
+
+	/**
+	 * Defines a formatted date value.
+	 * @default undefined
+	 * @public
+	 */
+	@property({ noAttribute: true })
+	value?: DynamicDateRangeValue;
+
+	/**
+	 * Defines the open or closed state of the popover.
+	 * @public
+	 * @default false
+	 */
+	@property({ type: Boolean })
+	open = false;
+
+	/**
+	 * Defines the options.
+	 * @public
+	 * @default ""
+	 */
+	@property({ type: String })
+	options = "";
+
+    @property({ type: Object })
+    _currentOption?: IDynamicDateRangeOption;
+
+	@property({ type: Object })
+    currentValue?: DynamicDateRangeValue;
+
+	static optionsClasses: Map<string, new () => IDynamicDateRangeOption> = new Map();
+
+	optionsObjects: IDynamicDateRangeOption[] = [];
+
+	@query("[ui5-input]")
+	_input?: Input;
+
+	onBeforeRendering() {
+		if (!this.optionsObjects.length) {
+			const optionKeys = this.options.split(",").map(option => option.trim());
+			this.optionsObjects = optionKeys.map(option => {
+				const OptionClass = DynamicDateRange.getOptionClass(option);
+
+				if (OptionClass) {
+					const optionObject = new OptionClass();
+
+					return optionObject;
+				}
+
+				return undefined;
+			}).filter(optionObject => optionObject !== undefined);
+		}
+	}
+
+	get _optionsTitles(): Array<string> {
+		return this.optionsObjects.map(option => option.text);
+	}
+
+	get openIconName() {
+		return "appointment-2";
```

Comment:
the best practice is to have the icons imports in the TSX and use the string, exporeted by the icon module directly in the template